modules/http/http-curl-compression.c

```diff
@@ -87,4 +98,62 @@ int _zstd_string(GString *compressed, const GString *message)
 {
   //not yet implemented
   g_assert_not_reached();
+}
+
+int _deflate_type_compression(GString *compressed, const GString *message, const gint deflate_algorithm_type)
+{
+  z_stream _compress_stream = {0};
+  gint err;
+
+  _compress_stream.data_type = Z_TEXT;
+
+  _compress_stream.next_in = (guchar *)message->str;
+  _compress_stream.avail_in = message->len;
+  _compress_stream.total_in = _compress_stream.avail_in;
+
+  g_string_set_size(compressed, (guint)(_compress_stream.avail_in * 1.1) + 22);
+
+  //Check buffer overrun
+  if(_compress_stream.avail_in != message->len)
+    {
+      return Z_BUF_ERROR;
+    }
+  _compress_stream.next_out = (guchar *)compressed->str;
+  _compress_stream.avail_out = compressed->len;
+  _compress_stream.total_out = _compress_stream.avail_out;
+  if(_compress_stream.avail_out != compressed->len)
+    {
+      return Z_BUF_ERROR;
+    }
+
+  gint _wbits;
+  switch (deflate_algorithm_type)
+    {
+    case DEFLATE_TYPE_DEFLATE:
+      _wbits = _DEFLATE_WBITS_DEFLATE;
+      break;
+    case DEFLATE_TYPE_GZIP:
+      _wbits = _DEFLATE_WBITS_GZIP;
+      break;
+    default:
+      g_assert_not_reached();
+    }
```

Comment:
I think a single relatively short and unambiguous switch-case does not warrant it's own function, since it is only used once, does not hinder readability and does not obfuscate controll-flow. I, however, can be convinced of the of the opposite :)

Reply:
This function does 4 things as far as I can see:
  1. prepares the compression stream
  2. gets the compression related wbit size
  3. compresses the data
  4. converts result code

When someone will read this code in the future, they will not necessary be interested in all of the impementation details, maybe they are just interested about the high-level steps of this conversion (see the 4 points above). Maybe they will be interested in the prepare step, or the wbit size choosing after they check this function, and if those are moved to separate functions, the reader can jump to it and get into the details of only those things. This is generally endorsed by clean code principles.

In an ideal case this function would only call 4 other functions, one for each point I mentioned. We do not always conform to this of course, and the preference of the developer matters.

In this case I would split this function like I mentioned, this is my feeling/preference, but you can leave it as is if you want.

---

Probably an [optional] tag and this explanation should have been there in my original comment, sorry for that. :/

Reply:
No problem :). I might refactor it to make easier to understand

Reply:
Done